packages/builder/src/registry.ts

```diff
@@ -480,7 +480,36 @@ export class OnChainRegistry extends CannonRegistry {
       ],
     });
 
-    return (url as string) || null;
+    const response = await axios.post(
+      this.provider.transport.url,
+      {
+        jsonrpc: '2.0',
+        method: 'eth_call',
+        params: [
+          {
+            to: this.contract.address,
+            data: encodedData,
+          },
+        ],
+      },
+      {
+        headers: {
+          'Content-Type': 'application/json',
+        },
+      }
+    );
+
+    if (response.data && response.data.error) {
+      throw response.data.error.message;
```

Comment:
Is the `message` prop a string? in that case it could be thrown as an error to have a full stack trace:
```suggestion
      throw new Error(response.data.error.message);
```